packages/graphql/tests/integration/directives/auth/jwks-endpoint.int.test.ts

```diff
@@ -0,0 +1,348 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Driver } from "neo4j-driver";
+import { graphql } from "graphql";
+import { generate } from "randomstring";
+import createJWKSMock, { JWKSMock } from "mock-jwks";
+import supertest from "supertest";
+import Koa from "koa";
+import Router from "koa-router";
+import jwt from "koa-jwt";
+import jwksRsa from "jwks-rsa";
+import neo4j from "../../neo4j";
+import { Neo4jGraphQL } from "../../../../src/classes";
+
+describe("https://github.com/neo4j/graphql/issues/564", () => {
+    let jwksMock: JWKSMock;
+    let server: any;
+    let request: any;
+    let driver: Driver;
+
+    beforeAll(async () => {
+        driver = await neo4j();
+    });
+
+    afterAll(async () => {
+        await driver.close();
+    });
+
+    beforeEach(() => {
+        ({ jwksMock, server, request } = createContext());
+    });
+
+    afterEach(async () => {
+        await tearDown({ jwksMock, server });
+    });
+
+    test("should not get access without correct token", async () => {
+        jwksMock.start();
+        const { status } = await request.get("/");
+        expect(status).toEqual(401);
+    });
+
+    test("should get access with mock token when jwksMock is running", async () => {
+        jwksMock.start();
+        const accessToken = jwksMock.token({
+            aud: "private",
+            iss: "master",
+        });
+        const { status } = await request.get("/").set("Authorization", `Bearer ${accessToken}`);
+        expect(status).toEqual(200);
+    });
+
+    test("should not get access with mock token when jwksMock is not running", async () => {
+        // Now we do not intercept queries. The queries of the middleware for the JKWS will
+        // go to the production server and the local key will be invalid.
+        const accessToken = jwksMock.token({
+            aud: "private",
+            iss: "master",
+        });
+        const { status } = await request.get("/").set("Authorization", `Bearer ${accessToken}`);
+        expect(status).toEqual(401);
+    });
+
+    test("Another example with a non-auth0-style jkwsUri", async () => {
+        const jwksMockCustom = createJWKSMock("https://myAuthTest.auth0.com", "/protocol/openid-connect/certs");
+
+        // We start our app.
+        const serverCustom = createApp({
+            jwksUri: "https://myAuthTest.auth0.com/protocol/openid-connect/certs",
+        }).listen();
+
+        const requestCustom = supertest(serverCustom);
+
+        jwksMockCustom.start();
+        const accessToken = jwksMockCustom.token({
+            aud: "private",
+            iss: "master",
+        });
+        const { status } = await requestCustom.get("/").set("Authorization", `Bearer ${accessToken}`);
+        await tearDown({ jwksMock: jwksMockCustom, server: serverCustom });
+        expect(status).toEqual(200);
+    });
```

Comment:
These particular tests don't appear to be testing anything of this project at all?

Reply:
You are correct, these test the settings of the jwksMock server itself. I kept it to ensure the server is up and running as intended before the actual tests. Mostly because I didn't how a local mock server will react in your GitHub CI/CD on-prem tests. But if you want to clean up clutter, we can remove it... your call